modules/registration/registration_api.rb

```diff
@@ -0,0 +1,11 @@
+require 'registration/proxy_request'
+
+class Proxy::Registration::Api < ::Sinatra::Base
+  get '/' do
+    Proxy::Registration::ProxyRequest.new.global_register(request)
```

Comment:
I would consider not having `global_register` and `host_register` as separate methods and inline them here. The reason is that currently you don't do anything with the response code. However, I think it is very useful for responses from the Smart Proxy to also have a correct response code. For example, on a timeout (which you currently don't capture) you should probably return a HTTP 503 code.

There may also be other HTTP errors, such as invalid certificates etc. Since the expected use of this API is to pipe it directly to bash, it's crucial you capture any exception and always return something valid but with the appropriate context.

It's also useful to have logging when a request fails.

Reply:
> I would consider not having global_register and host_register as separate methods and inline them here. The reason is that currently you don't do anything with the response code. 

I do, here: https://github.com/theforeman/smart-proxy/pull/768/files#diff-45f33bea51f0f06fd02907550e854ae2R37-R42
Or should I do it in different way? If yes could you show me how do you want to do it?

> There may also be other HTTP errors, such as invalid certificates etc. Since the expected use of this API is to pipe it directly to bash, it's crucial you capture any exception ...

Can you show me exactly how you want me to do it, so we are both on same boat.

Reply:
That is only the body you deal with for incoming requests. Clients for the Smart Proxy don't see those, do they? The fact `assert last_response.ok?` in the tests pass indicate that it returns a HTTP 200 instead of 500 (or similar).

Reply:
>  in the tests pass indicate that it returns a HTTP 200 instead of 500 (or similar)

Yeah but it is used in templates, so if there is an error we render it like this:
```
echo '<message>'
exit 1
```
So the script will exit anyway, no matter what status code is.

Reply:
But if you pass the HTTP status code, you at least know that it failed. You can see this in proxy.log and it's just a good practice.

For example:

```
2020-10-01T16:49:29 bf9e90fb [I] Started GET /features 
2020-10-01T16:49:29 bf9e90fb [I] Finished GET /features with 200 (24.44 ms)
```

If you see there are HTTP 5xx status codes, you know there's a problem. This can be used by various logging tools as well. Out of experience I know that debugging such failures can be painful and that's why I strongly feel that this is important.